core/src/main/java/org/wso2/testgrid/core/configchangeset/ConfigChangeSetExecutorUnix.java

```diff
@@ -0,0 +1,242 @@
+     *
+     * @param testPlan  The test plan
+     * @return          True if execution success. Else, false
+     */
+    @Override
+    public boolean initConfigChangeSet(TestPlan testPlan) {
+        try {
+            URL configChangeSetRepoPath = new URL(testPlan.getConfigChangeSetRepository());
+            String filePath = configChangeSetRepoPath.getPath();
+            String fileName = filePath.substring(filePath.lastIndexOf('/') + 1);
+            List<String> initShellCommand = new ArrayList<>();
+            initShellCommand.add("mkdir repos");
+            initShellCommand.add("cd repos && curl -LJO " +
+                    testPlan.getConfigChangeSetRepository() + "/archive/master.tar.gz   &>/dev/null");
+            initShellCommand.add("cd repos && tar xvzf " + fileName + "-master.tar.gz  &>/dev/null");
+            initShellCommand.add("chmod -R 755 repos/" + fileName + "-master/config-sets/ &>/dev/null");
+            return applyShellCommandOnAgent(testPlan, initShellCommand);
+        } catch (MalformedURLException e) {
+            logger.warn("Error parsing scenario repository path for ".
+                    concat(testPlan.getConfigChangeSetRepository()), e);
+            return false;
+        }
+    }
+
+    /**
+     * Revert back changes did in initConfigChangeSet
+     *
+     * @param testPlan  The test plan
+     * @return          True if execution success. Else, false
+     */
+    @Override
+    public boolean revertConfigChangeSet(TestPlan testPlan) {
+        List<String> revertShellCommand = new ArrayList<>();
+        revertShellCommand.add("rm -rf repos");
+        return applyShellCommandOnAgent(testPlan, revertShellCommand);
+    }
+
+    /**
+     * Send an array of shell command to an agent
+     *
+     * @param testPlan          The test plan
+     * @param initShellCommand  List of shell command
+     * @return  True if execution success. Else, false
+     */
+    private boolean applyShellCommandOnAgent(TestPlan testPlan, List<String> initShellCommand) {
+        String tinkererHost = ConfigurationContext.getProperty(
+                ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_REST_BASE_PATH);
+        // Get authentication detail from config.properties and encode with BASE64
+        String authenticationString = ConfigurationContext.getProperty(
+                ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_USERNAME) + ":" +
+                ConfigurationContext.getProperty(
+                        ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_PASSWORD);
+        String authenticationToken = "Basic " + Base64.getEncoder().encodeToString(
+                authenticationString.getBytes(StandardCharsets.UTF_8));
+        boolean executionStatus = true;
+        String agentLink = tinkererHost + "test-plan" + TestGridConstants.FILE_SEPARATOR + testPlan.getId() +
+                TestGridConstants.FILE_SEPARATOR + "agents";
```

Comment:
What you are building here is a URL. URLs has "/" forward-slash as the separator always, irrespective of the OS. 
You could rather use java.net.URL to build the agentLink url.